packages/firestore/src/core/query.ts

```diff
@@ -390,10 +390,39 @@ export class Query {
   }
 }
 
-export interface Filter {
-  matches(doc: Document): boolean;
-  canonicalId(): string;
-  isEqual(filter: Filter): boolean;
+export abstract class Filter {
+  abstract matches(doc: Document): boolean;
+  abstract canonicalId(): string;
+  abstract isEqual(filter: Filter): boolean;
+
+  /**
+   * Creates a filter based on the provided arguments.
+   */
+  static create(
+    field: FieldPath,
+    op: RelationOp,
+    value: FieldValue
+  ): Filter {
+    if (value.isEqual(NullValue.INSTANCE)) {
+      if (op !== RelationOp.EQUAL) {
+        throw new FirestoreError(
+          Code.INVALID_ARGUMENT,
+          'Invalid query. You can only perform equals ' + 'comparisons on null.'
```

Comment:
Do you mind removing the extraneous string concatenation?

Reply:
Done.